selftests/functional/test_output.py

```diff
@@ -319,6 +363,21 @@ def test_show(self):
             self.assertEqual(expected_job_id_number, job_id_number)
             self.assertEqual(expected_bin_true_number, bin_true_number)
 
+    @skipUnlessPathExists('/bin/true')
+    @skipUnlessPathExists('/bin/false')
+    def test_avoid_output_on_non_test_task(self):
+        """
+        Checks if `core.show` is respected in different configurations.
+        """
+        with script.Script(os.path.join(self.tmpdir.name,
+                                        'test_avoid_output_non_test_tasks.py'),
+                           AVOID_NON_TEST_TASKS) as test:
+            result = process.run(test.path)
```

Comment:
Either decorate this test with a `@fail_on(CmdError...` or do a `process.run(..., ignore_status=True)` paired with a `self.assertEqual(result.exit_code, AVOCADO_ALL_OK)` so that a failure will be signaled as `FAIL` and not as an `ERROR`.